internal/jobcontainers/process.go

```diff
@@ -95,7 +96,13 @@ func (p *JobProcess) Signal(ctx context.Context, options interface{}) (bool, err
 	}
 
 	if err := signalProcess(uint32(p.cmd.Pid()), signal); err != nil {
-		return false, errors.Wrap(err, "failed to send signal")
+		// This could be a case of the process exited between us calling p.exited()
+		// above and here. Access is denied appears to be returned if you try and open a remote
+		// thread for a process that has exited just like TerminateProcess. We should have permission
+		// to open the thread, so this error should only occur for the "process is dead" case.
+		if !errors.Is(err, windows.ERROR_ACCESS_DENIED) {
+			return false, fmt.Errorf("failed to send signal %d: %w", signal, err)
+		}
```

Comment:
Can remove this after https://github.com/microsoft/hcsshim/pull/1447 lands

Reply:
Can you add a TODO in case this is merged before #1447?

Reply:
I merged the other, so I'll push an update here

Reply:
Removing after rebasing to grab Gabriel's fix